examples/kdigest.c

```diff
@@ -0,0 +1,292 @@
+/* SPDX-License-Identifier: MIT */
+/*
+ * link-cp based proof-of-concept for doing file digests with linked SQEs using
+ * the kernel's AF_ALG API. Needs a bit of error handling and short read love.
+ */
+#include <stdio.h>
+#include <fcntl.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <assert.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <linux/if_alg.h>
+#include "liburing.h"
+
+#define QD	64
+#define BS	(32*1024)
+
+struct io_data {
+	size_t offset;
+	int index;
+	struct iovec iov;
+};
+
+static int infd, outfd;
+static unsigned inflight;
+
+static int setup_context(unsigned entries, struct io_uring *ring)
```

Comment:
Why the function does even exist but not inlined?

Reply:
I think this is carryover from link-cp. I'll look at simplifying this next round.